src/common/dsp/vembertech/basic_dsp.h

```diff
@@ -54,6 +54,15 @@ inline void i152float_block(short *s, float *f, int n)
     }
 }
 
+inline void i162float_block(short *s, float *f, int n)
+{
+    const float scale = 1.f / (16384.f * 2);
+    for (int i = 0; i < n; i++)
+    {
+        f[i] = (float)s[i] * scale;
```

Comment:
Maybe `std::clamp(x, -1, 1)`?
I think you either need to clamp or assert that the input value is in the expected range.
You could do both as well that may catch programming error.

Reply:
So the reason I don't clip it here is that the window oscillator has a wee bit of headroom but yeah i'm not sure where we want the clipping, but i think we don't want to *move* the clipping at this point.